File: 3653. xor-after-range-multiplication-queries-i.py
#translated using AI
class Solution:
    def xorAfterQueries(self, nums, queries):
        MOD = 10**9 + 7

        for l, r, k, v in queries:
            i = l
            while i <= r:
                nums[i] = (nums[i] * v) % MOD
                i += k

        ans = 0
        for num in nums:
            ans ^= num

        return ans

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    nums1 = [1,1,1]
    queries1 = [[0,2,1,4]]
    output1 = sol.xorAfterQueries(nums1, queries1)
    print(f"Example 1: nums=[1,1,1], queries=[[0,2,1,4]]")
    print(f"Output: {output1}")
    print(f"Expected: 4")
    print()
    
    # Example 2
    nums2 = [2,3,1,5,4]
    queries2 = [[1,4,2,3],[0,2,1,2]]
    output2 = sol.xorAfterQueries(nums2, queries2)
    print(f"Example 2: nums=[2,3,1,5,4], queries=[[1,4,2,3],[0,2,1,2]]")
    print(f"Output: {output2}")
    print(f"Expected: 31")

File: 3653. xor-after-range-multiplication-queries-i.java
class Solution {
    int MOD = 1_000_000_007;

    public int xorAfterQueries(int[] nums, int[][] queries) {
        for (int[] query : queries) {
            int l = query[0];
            int r = query[1];
            int k = query[2];
            int v = query[3];

            for (int i = l; i <= r; i += k) {
                nums[i] = (int)((1L * nums[i] * v) % MOD);
            }
        }

        int ans = 0;
        for (int num : nums) {
            ans ^= num;
        }

        return ans;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        int[] nums1 = {1,1,1};
        int[][] queries1 = {{0,2,1,4}};
        int output1 = sol.xorAfterQueries(nums1, queries1);
        System.out.println("Example 1: nums=[1,1,1], queries=[[0,2,1,4]]");
        System.out.println("Output: " + output1);
        System.out.println("Expected: 4");
        System.out.println();
        
        // Example 2
        int[] nums2 = {2,3,1,5,4};
        int[][] queries2 = {{1,4,2,3},{0,2,1,2}};
        int output2 = sol.xorAfterQueries(nums2, queries2);
        System.out.println("Example 2: nums=[2,3,1,5,4], queries=[[1,4,2,3],[0,2,1,2]]");
        System.out.println("Output: " + output2);
        System.out.println("Expected: 31");
    }
}